### PR TITLE
Integrate gps driver

### PR DESCRIPTION
**Pull request by MCLiii**
Automatically find lte modem and parse gps data, only works on linux, require sudo privilege.

### CODE REVIEW

**Comment by MCLiii** [comment]
Sorry for the lengthy PR, you can ignore anything added to the 3rdparty directory.

File: DataProcessor/DataUnpacker.cpp [hunk 1]
```diff
@@ -80,6 +80,13 @@ DataUnpacker::DataUnpacker(QObject *parent) : QObject(parent)
                 cell_group_voltages_end = dataCount;
             }
             cell_group_voltages.push_back(0);
+        } else if (name == "lat") {
+            qDebug() << "aaaasadasdasdsa";
```
Comment (jackwherry) [inline DataProcessor/DataUnpacker.cpp]: most normal debug message

File: gps/gps.cpp [hunk 2]
```diff
@@ -0,0 +1,190 @@
+#include "gps.h"
+#include <string>
+#include <stdio.h>
+#include <filesystem>
+#include <sstream>
+#include <iostream>
+
+using namespace std;
```
Comment (jackwherry) [inline gps/gps.cpp:9]: Is this a style we want to use? https://stackoverflow.com/questions/1452721/whats-the-problem-with-using-namespace-std

**Review by TheLawrus — APPROVED** [review]
lgtm

**Comment by MCLiii** [comment]
@TheLawrus Let's use rebase instead of merge to keep history straight

**Pull request merged**